docs/fundamentals/runtime-libraries/system-reflection-emit-assemblybuilder.md

```diff
@@ -58,38 +58,51 @@ The <xref:System.Reflection.Emit.AssemblyBuilder.Save%2A?displayProperty=nameWit
-    ILGenerator il = meb.GetILGenerator();
-    il.Emit(OpCodes.Ldarg_0);
-    il.Emit(OpCodes.Ldarg_1);
-    il.Emit(OpCodes.Add);
-    il.Emit(OpCodes.Ret);
-
-    tb.CreateType();
-
-    using var stream = new MemoryStream();
-    ab.Save(stream);  // or pass filename to save into a file
-    stream.Seek(0, SeekOrigin.Begin);
-    Assembly assembly = AssemblyLoadContext.Default.LoadFromStream(stream);
-    MethodInfo method = assembly.GetType("MyType").GetMethod("SumMethod");
-    Console.WriteLine(method.Invoke(null, new object[] { 5, 10 }));
-}
-```
-
-In order to set entry point for an executable and/or set other options for the assembly file you could to call the `public MetadataBuilder GenerateMetadata(out BlobBuilder ilStream, out BlobBuilder mappedFieldData)` method and use the populated metadata for generating the assembly with desired the options, for example:
+      ILGenerator il = meb.GetILGenerator();
+      il.Emit(OpCodes.Ldarg_0);
+      il.Emit(OpCodes.Ldarg_1);
+      il.Emit(OpCodes.Add);
+      il.Emit(OpCodes.Ret);
+
+      tb.CreateType();
+
+      using var stream = new MemoryStream();
+      ab.Save(stream);  // or pass filename to save into a file
+      stream.Seek(0, SeekOrigin.Begin);
+      Assembly assembly = AssemblyLoadContext.Default.LoadFromStream(stream);
+      MethodInfo method = assembly.GetType("MyType").GetMethod("SumMethod");
+      Console.WriteLine(method.Invoke(null, new object[] { 5, 10 }));
+  }
+  ```
+
+- If `Reflection.Emit` is used to generate an assembly that targets a specific TFM, open the reference assemblies for the given TFM using `MetadataLoadContext` and use the value of the [MetadataLoadContext.CoreAssembly](xref:System.Reflection.MetadataLoadContext.CoreAssembly) property for `coreAssembly`. This value allows the generator to run on one .NET runtime version and target a different .NET runtime version. Note that, you should use types returned by the MetadataLoadContext instance when referencing core types. For example, instead of `typeof(object)`, you should lookup `System.Object` type in `MetadataLoadContext.CoreAssembly` by name:
+
+  ```csharp
+  public void CreatePersistedAssemblyBuilderCoreAssemblyWithMetadataLoadContext(string refAssembliesPath)
+  {
+      PathAssemblyResolver resolver = new PathAssemblyResolver(Directory.GetFiles(refAssembliesPath, "*.dll"));
+      using MetadataLoadContext context = new MetadataLoadContext(resolver);
+      Assembly coreAssembly = context.CoreAssembly;
+      PersistedAssemblyBuilder ab = new PersistedAssemblyBuilder(new AssemblyName("MyDynamicAssembly"), coreAssembly);
+      TypeBuilder typeBuilder = ab.DefineDynamicModule("MyModule").DefineType("Test", TypeAttributes.Public);
+      MethodBuilder methodBuilder = typeBuilder.DefineMethod("Method", MethodAttributes.Public, coreAssembly.GetType(typeof(int).FullName), Type.EmptyTypes);
```

Comment:
```suggestion
      MethodBuilder methodBuilder = typeBuilder.DefineMethod("Method", MethodAttributes.Public, coreAssembly.GetType("System.Int32"), Type.EmptyTypes);
```
To make it more clear that the MetadataLoadContext-based system is not associated with the runtime-based system in any way.